### ISSUE TITLE
Unable to create dbformysql.PrivateEndpointConnection - cannot create already existing resource

### ISSUE THREAD

**Issue by clstokes** [pulumi/pulumi-azure-native#808]
No matter what name or properties I set for a `dbformysql.PrivateEndpointConnection` resource, I always get the error `cannot create already existing resource`.  If I `az resource list --location westus --resource-group <resource_group>`, I only see the `Microsoft.DBForMySQL/servers` resource and nothing else (based on the code in the gist below).

## Expected behavior

Should be able to create the resource.

## Current behavior

It fails:

```
p-azure-native-sql-privateconn % pulumi up -y --skip-preview
Updating (dev)

View Live: https://app.pulumi.com/clstokes/p-azure-native-sql-privateconn/dev/updates/3

     Type                                                  Name                                Status                  Info
 +   pulumi:pulumi:Stack                                   p-azure-native-sql-privateconn-dev  **creating failed**     1 error
 +   ├─ azure-native:resources:ResourceGroup               main                                created                 
 +   ├─ azure-native:dbformysql:Server                     main                                created                 
 +   └─ azure-native:dbformysql:PrivateEndpointConnection  main                                **creating failed**     1 error
 
Diagnostics:
  pulumi:pulumi:Stack (p-azure-native-sql-privateconn-dev):
    error: update failed
 
  azure-native:dbformysql:PrivateEndpointConnection (main):
    error: cannot create already existing resource '/subscriptions/32b9cb2e-69be-4040-80a6-02cd6b2cc5ec/resourceGroups/main205a2a7e/providers/Microsoft.DBforMySQL/servers/mainb36d3358/privateEndpointConnections/main'
 
Resources:
    + 3 created

Duration: 1m27s
```

## Steps to reproduce

1. `git clone https://gist.github.com/d9cc501432cdc16f4af29dd61b946bb5 p-azure-native-sql-privateconn`
2. `cd p-azure-native-sql-privateconn`
3. `pulumi stack init dev`
4. `pulumi up -y --skip-preview`

**Comment by mikhailshilkov** [comment]
@clstokes Could you run it with debug options and find the HTTP request and response for that resource operation?

**Comment by mikhailshilkov** [comment]
Nevermind, I got it from your repro:

```
HTTP Request Begin GET https://management.azure.com/subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/main003664c0/providers/Microsoft.DBforMySQL/servers/main3028679a/privateEndpointConnections/main?api-version=2018-06-01 
    Empty body
HTTP Request End GET https://management.azure.com/subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/main003664c0/providers/Microsoft.DBforMySQL/servers/main3028679a/privateEndpointConnections/main?api-version=2018-06-01
HTTP Response Begin GET [https://management.azure.com/subscriptions/0282681f-7a9e-424b-80b2-96babd57a8a1/resourceGroups/main003664c0/providers/Microsoft.DBforMySQL/servers/main3028679a/privateEndpointConnections/main?api-version=2018-06-01 
    HTTP/1.1 204 No Content
    Cache-Control: no-cache
    Date: Sun, 02 May 2021 09:58:30 GMT
    Expires: -1
    Pragma: no-cache
    Server: Microsoft-HTTPAPI/2.0
    Strict-Transport-Security: max-age=31536000; includeSubDomains
    X-Content-Type-Options: nosniff
    X-Ms-Correlation-Request-Id: 48d880aa-c6f1-4d9b-a2cf-fa95b7b24ec6
    X-Ms-Ratelimit-Remaining-Subscription-Reads: 11999
    X-Ms-Request-Id: 58a46776-cd38-4736-b1fb-16b51cfc169d
    X-Ms-Routing-Request-Id: WESTEUROPE:20210502T095831Z:48d880aa-c6f1-4d9b-a2cf-fa95b7b24ec6
```

**Comment by mikhailshilkov** [comment]
We'll relax the check to success on 204 for GET/POST resources. The check was introduced for HEAD-based resources in https://github.com/pulumi/pulumi-azure-native/pull/400/commits/edadaad0b1a4c55949ce9b1ab57a62d5c0dc5d70 but it's apparently too wide.

**Comment by mikhailshilkov** [comment]
I have a fix for this particular problem but then the actual PUT request to create the resource returns the same 204 which fails the creation (which is reasonable, as 204 is not listed as a valid response in [the spec](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2018-06-01/PrivateEndpointConnections.json#L119). 

I believe you need to set up the actual private endpoint connection with all the networking resources (as described [here](https://docs.microsoft.com/en-us/azure/private-link/create-private-endpoint-template) for a SQL Server). Maybe that will fix the error if everything is in place?

**Comment by mikhailshilkov** [comment]
@clstokes the reported error is fixed in 1.4.0. As I mention above, your repro will still fail at a later stage.

**Comment by mikhailshilkov** [comment]
The original problem is fixed, the follow-up is tracked in #867
